### ISSUE TITLE
Picom-Git update leads to severe lag and loss of control in Arch Linux Systems

### ISSUE THREAD

**Issue by raven2cz** [yshui/compton#1077]
I have noticed significant system lag since yesterday, 06/17/2023. It stops responding to events and becomes very heavily slowed down, making it unmanageable. This concerns picom-git on Arch Linux. There was no problem before, it is purely due to an update a few days old.

However, this is happening on all my desktops and laptops. They include AMD, Nvidia, and Intel graphics cards, so it is independent of the graphics, it must be some critical issue. Please set a high priority for this and make repairs, thank you.

My config:
https://github.com/raven2cz/dotfiles/blob/main/.config/picom/picom.conf

**Comment by absolutelynothelix** [comment]
try `--no-frame-pacing`. if it helps see #1072.

**Comment by raven2cz** [comment]
> try `--no-frame-pacing`. if it helps see #1072.

After half a day of testing, I would say that it seems to be the right solution. After adjusting that parameter, certain parts that were causing the issue are presumably not being utilized, and the system appears to be stable. I will observe it for a few more days. Does this mean that the problem is resolved, or will they need to fix something? The issue is that without that parameter, picom is unusable with awesomewm.

**Comment by yshui** [comment]
@raven2cz can you run picom with `--log-level trace`, and **without** `--no-frame-pacing`, wait until the problem occurs, and upload the logs here? thanks.

**Comment by raven2cz** [comment]
> @raven2cz can you run picom with `--log-level trace`, and **without** `--no-frame-pacing`, wait until the problem occurs, and upload the logs here? thanks.

As of yesterday, I have been unable to reproduce the issue on three different platforms: AMD, NVIDIA, and Intel. It's possible that this has been resolved, or it may take some time to simulate. I will continue testing over the next few days.

I have noticed a significant issue, primarily with **Intel**. Without using the `--no-frame-pacing` parameter, the animations are slow and exhibit tearing. However, with the `--no-frame-pacing` parameter, the performance is almost on par with what it was a few days ago, though it is still somewhat **slower**.
